packages/@uppy/core/src/Uppy.js

```diff
@@ -1573,4 +1570,14 @@ class Uppy {
   }
 }
 
+// todo come up with a better name or remove the need for two different generators
+export function generateFileId2 (file) {
```

Comment:
Why do we need a new file ID generator?

Reply:
we already had 2 id generators. i just pulled out the code of one of them into a function. i don't know why we have 2

Reply:
Perhaps we can run a quick test without it? Generally code like this is not revisited and sticks around.

Reply:
I'm scared to change code that I don't understand how works. I don't want to break some corner case. I think we should do that in a different PR. Or maybe @arturi has some insights as to why there are two different methods for generating an ID

Reply:
I just read the `checkAndCreateFileStateObject` function and failed to see two ways to generate IDs there. Could you point me to version number 1 and 2?

Reply:
one is here:

https://github.com/transloadit/uppy/blob/e6c9a4690cbcb6949bfe057856ca753efc02b98f/packages/%40uppy/provider-views/src/View.js#L26

other is here:

https://github.com/transloadit/uppy/blob/e6c9a4690cbcb6949bfe057856ca753efc02b98f/packages/%40uppy/core/src/Uppy.js#L461

Reply:
Ah, found that too, but ignored since it’s in a different file. That calls to the same `import generateFileID from '@uppy/utils/lib/generateFileID'` though, so it’s not like it’s defined twice?

I think the reason fileID is generated there is because non-remote, regular files expected by Uppy are `{ data, name ... }`, but remote file object has no data, and sometimes no name, but we could check for that in core `checkAndCreateFileStateObject`, like we do with the File object.

It seems to me the `id` might be more correctly generated in `View`, but then is ignored in `checkAndCreateFileStateObject` anyway, so we can remove it indeed?

Reply:
so you think we can remove providerFileToId completely and instead use `generateFileId2` in its place (but rename it to something better)?

Reply:
Just revert the Core change, yes. Or leave this all entirely out of this pr.

Reply:
the thing is that i need this code from core:
```
    const fileType = getFileType(file)

    const fileID = generateFileID({
      ...fileDescriptor,
      type: fileType,
    })
```

...inside providerView in order to make sure that the file is not already added (it needs the ID to check that). I didn't want to copy-paste that code, so I instead put it in a function (hence generateFileId2)

Reply:
Could you just call the util generateFileID directly and feed it the file obj?

Reply:
no, I think that doesn't work, because it will generate a different ID than the once generated in core. Thus the equality will not work as expected when comparing to see if a file is already added (ID will be different for the same file)

Reply:
Can't we feed it the same file data? There's no random uuid or such, why would it be different?

Reply:
beucase `generateFileID` uses `type`:

https://github.com/transloadit/uppy/blob/57cc08eae594b48c68263e433d36c52a6c85d858/packages/%40uppy/utils/src/generateFileID.js#L30

so if type is different, the ID returned from `generateFileID` will be different

Reply:
True, why would the type be different? :see_no_evil: The file is added initially from ProviderView, so whatever type it gets from Gdrive or similar, that's what we pass?

Reply:
one of the functions sets `type: file.mimetype`, the other sets `type: getFileType(file)`, so they could be different, no?

Reply:
and the other one sets `data` and `name` to different values too, which could affect the ID too:
```
  providerFileToId (file) {
    return generateFileID({
      data: file,
      name: file.name || file.id,
      type: file.mimetype,
    })
  }
```

Reply:
Yes, but not for the same file added from the same source. If you add X from Google Drive to Uppy, then trying to add X again will result in the same file id being generated and file being ignored as duplicate. 

Currently Core just ignores the `id` passed to `uppy.addFile()` — it generates its own. So I guess there’s no point in even passing it in — just commented out `providerFileToId` — remote files are still added correctly.

Reply:
Ok, if you think we can remove providerFileToId, let's do that. let's wait for your form PR if you want to merge that first, before doing too many changes here.